sdk/cxx/include/sawtooth/global_state.h

```diff
@@ -0,0 +1,77 @@
+/* Copyright 2017 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+------------------------------------------------------------------------------*/
+#pragma once
+
+#include <map>
+#include <memory>
+#include <string>
+
+#include "sawtooth/message_stream.h"
+
+namespace sawtooth {
+
+// Provides access to the global state
+class GlobalState {
+ public:
+    typedef std::pair<std::string, std::string> KeyValue;
+ public:
+    // Takes the messages stream that it will use to communicate with the
+    // Validator to retrieve the current state. It is not expected that this
+    // will be called directly from a Transaction Processor. An instance will
+    // be provided to transaction handlers when a transaction is being
+    // processed.
+    explicit GlobalState(const MessageStreamPtr& message_stream,
+        const std::string& context_id);
+    virtual ~GlobalState() {};
+
+    // Retrieve a single value from global state. If you are retrieving
+    // multiple values, it is encouraged to use the batch get overload defined
+    // below.
+    bool Get(std::string* out_value, const std::string& address);
```

Comment:
Gotcha - it seems that their standard argues for taking pointer instead of reference, and I can buy that.  I would still vote for the third then - return the std::string instead - since they don't seem to address it (although didn't read the entire spec and just searched for terms I thought would match the situation of deciding between taking a pointer for output versus returning a single value).